chainer/__init__.py

```diff
@@ -89,6 +93,45 @@ def get_function_hooks():
     return ret
 
 
+def _load_array_types():
+    global _array_types
+    global _cpu_array_types
+    if _array_types is None:
+        _array_types = [numpy.ndarray]
+        _cpu_array_types = [numpy.ndarray]
+
+        if cuda.available:
+            _array_types.append(cuda.ndarray)
+
+        if backends.intel64.is_ideep_available():
+            _array_types.append(backends.intel64.ideep.mdarray)
+            _cpu_array_types.append(backends.intel64.ideep.mdarray)
+
+        _array_types = tuple(_array_types)
+        _cpu_array_types = tuple(_cpu_array_types)
+
+
+def get_array_types():
+    _load_array_types()
+    return _array_types
+
+
+def get_cpu_array_types():
+    _load_array_types()
+    return _cpu_array_types
+
+
+def is_arrays_compatible(arrays):
```

Comment:
I neither have a strong opinion, but do *weakly* prefer explicit tuple as the current PR.
Perhaps it's better to hear other people's opinions.

Reply:
I personally don't that they need to be consistent. `get_array_module` may very well accept a single array while `is_arrays_compatible` is meant to compare multiple instances.

Reply:
Ah, that's a good point. Actually `is_arrays_comparible` can take a single array, but conceptually, your point may make sense. (The reason why I want to make it a tuple is more pragmatic, though: to allow future extension).

Reply:
I see. Then I feel we can leave it as is. My understanding is also that this function is mainly for internal usage (as it's not documented) in which case we could reconsider the interface at a later time.

Reply:
OK. My initial motivation was that I want to avoid confusion between usage of two similar functions (which accepts tuples and which accepts va-args) while coding, even for core developers, but I think @hvy's comment makes sense. Let's leave as it is.

Reply:
(NumPy is indeed confusing in that sense...)